community/kernel/src/main/java/org/neo4j/graphdb/factory/GraphDatabaseBuilder.java

```diff
@@ -39,8 +39,13 @@
  */
 public class GraphDatabaseBuilder
 {
+    @Deprecated
     public interface DatabaseCreator
     {
+        /**
+         * This method will no longer accept a map in 4.0, but a Config object instead.
+         */
+        @Deprecated
         GraphDatabaseService newDatabase( Map<String, String> config );
```

Comment:
Not exactly. That would break the implementation of anyone implementing the interface. To be non-breaking the new method would have to have a default implementation which delegates to the old method. But good idea nonetheless! See changes.